internal/pkg/manifest/validate.go

```diff
@@ -2050,29 +2054,40 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+type containerNameAndProtocol struct {
+	containerName     string
+	containerProtocol string
+}
+
 func validateExposedPorts(opts validateExposedPortsOpts) error {
-	containerNameFor := make(map[uint16]string)
-	populateMainContainerPort(containerNameFor, opts)
-	if err := populateSidecarContainerPortsAndValidate(containerNameFor, opts); err != nil {
+	portExposedTo := make(map[uint16]containerNameAndProtocol)
+
+	if err := populateAndValidateSidecarContainerPorts(portExposedTo, opts); err != nil {
 		return err
 	}
-	if err := populateALBPortsAndValidate(containerNameFor, opts); err != nil {
+	if err := populateAndValidateALBPorts(portExposedTo, opts); err != nil {
 		return err
 	}
-	if err := populateNLBPortsAndValidate(containerNameFor, opts); err != nil {
+	if err := populateAndValidateNLBPorts(portExposedTo, opts); err != nil {
+		return err
+	}
+	if err := populateAndValidateMainContainerPort(portExposedTo, opts); err != nil {
 		return err
 	}
 	return nil
 }
 
-func populateMainContainerPort(containerNameFor map[uint16]string, opts validateExposedPortsOpts) {
+func populateAndValidateMainContainerPort(portExposedTo map[uint16]containerNameAndProtocol, opts validateExposedPortsOpts) error {
 	if opts.mainContainerPort == nil {
-		return
+		return nil
 	}
-	containerNameFor[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+
+	port := aws.Uint16Value(opts.mainContainerPort)
+	protocol := defaultProtocol
+	return validateAndPopulateExposedPortMapping(portExposedTo, port, protocol, opts.mainContainerName)
```

Comment:
may be passing the args directly?
```suggestion
	return validateAndPopulateExposedPortMapping(portExposedTo, aws.Uint16Value(opts.mainContainerPort), defaultProtocol, opts.mainContainerName)
```